File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/converter/SmartCardRecordTypeToWalletRecordTypeConverter.java
package com.worldventures.dreamtrips.wallet.domain.converter;

import com.worldventures.dreamtrips.modules.mapping.converter.Converter;
import com.worldventures.dreamtrips.wallet.domain.entity.record.RecordType;

import io.techery.mappery.MapperyContext;

public class SmartCardRecordTypeToWalletRecordTypeConverter implements Converter<io.techery.janet.smartcard.model.Record.CardType, RecordType> {

   @Override
   public Class<io.techery.janet.smartcard.model.Record.CardType> sourceClass() {
      return io.techery.janet.smartcard.model.Record.CardType.class;
   }

   @Override
   public Class<RecordType> targetClass() {
      return RecordType.class;
   }

   @Override
   public RecordType convert(MapperyContext mapperyContext, io.techery.janet.smartcard.model.Record.CardType cardType) {
      switch (cardType) {
         case CREDIT:
            return RecordType.CREDIT;
         case DEBIT:
            return RecordType.DEBIT;
         case PREFERENCE:
            return RecordType.PREFERENCE;
         case FINANCIAL:
         default:
            return RecordType.FINANCIAL;
      }
   }
}

File: wallet/src/main/java/com/worldventures/wallet/domain/converter/SmartCardRecordTypeToWalletRecordTypeConverter.kt
package com.worldventures.wallet.domain.converter

import com.worldventures.core.converter.Converter
import com.worldventures.wallet.domain.entity.SDKCardType
import com.worldventures.wallet.domain.entity.record.RecordType

import io.techery.mappery.MapperyContext

class SmartCardRecordTypeToWalletRecordTypeConverter : Converter<SDKCardType, RecordType> {

   override fun sourceClass() = SDKCardType::class.java

   override fun targetClass() = RecordType::class.java

   override fun convert(context: MapperyContext, source: SDKCardType): RecordType {
      return when (source) {
         SDKCardType.CREDIT -> RecordType.CREDIT
         SDKCardType.DEBIT -> RecordType.DEBIT
         SDKCardType.PREFERENCE -> RecordType.PREFERENCE
         else -> RecordType.FINANCIAL
      }
   }
}
